ros/processor/inventory_events_consumer.py

```diff
@@ -79,13 +86,45 @@ def host_delete_event(self, msg):
                 insights_id,
                 self.prefix
             )
-            rows_deleted = db.session.query(PerformanceProfile).filter(
-                PerformanceProfile.inventory_id == host_id
-            ).delete()
+            system_id_query = db.session.query(System.id).filter(System.inventory_id == host_id).subquery()
+            rows_deleted = db.session.query(PerformanceProfile) \
+                .filter(PerformanceProfile.system_id.in_(system_id_query))
+
             if rows_deleted > 0:
                 LOG.info(
                     'Deleted %d performance profile record(s) - %s',
                     rows_deleted,
                     self.prefix
                 )
             db.session.commit()
+
+    def host_create_update_events(self, msg):
+        """ Process created/updated message ( create system record, store new report )"""
+        self.prefix = "PROCESSING Create/Update EVENT"
+        with app.app_context():
+            print(msg)
+            self.process_system_details(msg)
+
+    def process_system_details(self, msg):
+        """ Store new system information (stale, stale_warning timestamp) and return internal DB id"""
+        host = msg['host']
+
+        account = get_or_create(
+            db.session, RhAccount, 'account',
+            account=host['account']
+        )
+
+        db.session.add(account)
+        db.session.flush()
```

Comment:
Here you might end up trying to save account two times in DB. Because in `get_or_create` you are already doing `session.add(instance)` and `session.flush()`